### PR TITLE
Fix add_general_registry

### LINKED ISSUE #27: Error cloning General registry using buildpkg v1.3.1
I was encountering this error when building in the CI. The solution was to roll back to `- uses: julia-actions/julia-buildpkg@v1.2.0` in CI.yml

image

### PR DESCRIPTION
**Pull request by IanButterworth**
Fixes #27 

Need to look into testing this. Evidently it wasn't tested in #25

### CODE REVIEW

**Comment by IanButterworth** [comment]
@SaschaMann @DilumAluthge I haven't figured out why this wasn't detected in tests here, givven that `julia-buildpkg` should be calling `add_general_registry` here too.. but perhaps worth fixing this quick then figuring out the test later

**Comment by DilumAluthge** [comment]
Presumably the `isfile` check below should check for either `General/Registry.toml` or `General.toml`?

File: add_general_registry.jl [hunk 1]
```diff
@@ -30,8 +30,12 @@ end
...
     @info("Attempting to clone the General registry")
-    general_registry_dir, registry_toml_file = general_registry_location()
+    general_registry_tarball, registry_toml_file = tarball_general_registry_location()
+    rm(general_registry_tarball; force = true, recursive = true)
+    rm(registry_toml_file; force = true, recursive = true)
+    general_registry_dir, registry_toml_file = cloned_general_registry_location()
     rm(general_registry_dir; force = true, recursive = true)
+    
     Pkg.Registry.add("General")
     isfile(registry_toml_file) || throw(ErrorException("the Registry.toml file does not exist"))
```
Comment (IanButterworth) [inline add_general_registry.jl:40]:
```suggestion
    if !isfile(registry_toml_file) && !isfile(general_registry_tarball)
        throw(ErrorException("the Registry.toml file does not exist"))
    end
```

**Comment by IanButterworth** [comment]
Something fishy is going on with the CI set up here. 

Prior to [`5070133` (#28)](https://github.com/julia-actions/julia-buildpkg/pull/28/commits/5070133edc63342549be0d2a07e04dc59ecf04f2) it seemed like the `add_general_registry.jl` file was being overwritten by `julia-runtest`.

However [`5070133` (#28)](https://github.com/julia-actions/julia-buildpkg/pull/28/commits/5070133edc63342549be0d2a07e04dc59ecf04f2) apparently fixed that, but only by adding a missing `do`.

I think it was just a race between which `julia-` step got installed first.

Renaming that file is the robust fix though, both here and `julia-runtest`

**Review by DilumAluthge — APPROVED** [review]

**Comment by SaschaMann** [comment]
As mentioned in julia-runtest, I don't understand why this would be necessary, `GITHUB_ACTION_PATH` points to the repo of the action, not some shared workspace. They shouldn't interfere with each other in any way. How does this cause issues?

**Comment by SaschaMann** [comment]
I'll merge it already to fix the bug but I'm really unsure what happened there and would like to figure that out :D

**Comment by IanButterworth** [comment]
If the overwrite theory is false I don't have a clue 😬

**Pull request merged**